File: config/deploy.rb
require 'yaml'

set :application, 'phpschool'
set :repo_url, 'git@github.com:php-school/phpschool.io.git'
set :branch, 'master'
set :deploy_to, '/var/www/phpschool.io/'
set :pty, true
set :log_level, :debug
set :keep_releases, 2

set :linked_dirs, %w{var/logs public/uploads}
set :linked_files, %w{public/workshops.json .env}
namespace :deploy do

  task :schema_update do
    on roles(:web) do |host|
      within release_path do
        execute('php', 'vendor/bin/doctrine', 'orm:schema-tool:update', '-f')
      end
    end
  end



  task :generate_blog do
      on roles(:web) do |host|
        within release_path do
            execute('php', 'bin/app', 'generate-blog')
        end
      end
    end

  after "deploy:finished", "deploy:schema_update"
  after "deploy:schema_update", "deploy:generate_blog"
end

File: deploy.php
<?php

namespace Deployer;

require 'recipe/common.php';

set('application', 'phpschool');
set('repository', 'git@github-phpschool:php-school/phpschool.io.git');
set('branch', 'master');

set('shared_files', ['.env', 'public/workshops.json']);
set('shared_dirs', ['var/logs', 'public/uploads']);
set('writable_dirs', ['var/logs', 'public/uploads']);
set('keep_releases', 5);

set('git_tty', false);
set('allow_anonymous_stats', false);

host('imp')
    ->setHostname('100.70.13.88')
    ->setRemoteUser('deploy')
    ->set('deploy_path', '/var/www/phpschool.io');

// Install composer dependencies for the new release.
task('deploy:vendors', function () {
    run('cd {{release_path}} && {{bin/composer}} install --no-dev --optimize-autoloader --no-interaction --no-progress');
})->desc('Install composer dependencies');

// Post-deploy application tasks (ported from the old Capistrano hooks).
task('deploy:clear-cache', function () {
    run('cd {{release_path}} && {{bin/php}} bin/app clear-cache');
})->desc('Clear the application cache');

task('deploy:schema-update', function () {
    run('cd {{release_path}} && {{bin/php}} vendor/bin/doctrine orm:schema-tool:update -f');
})->desc('Update the Doctrine schema');

task('deploy:generate-blog', function () {
    run('cd {{release_path}} && {{bin/php}} bin/app generate-blog');
})->desc('Generate the blog');

// Full deploy flow.
task('deploy', [
    'deploy:prepare',
    'deploy:vendors',
    'deploy:clear-cache',
    'deploy:schema-update',
    'deploy:generate-blog',
    'deploy:publish',
])->desc('Deploy phpschool.io');

// Unlock automatically if the deploy fails.
after('deploy:failed', 'deploy:unlock');
